### PR TITLE
[WFCORE-7031] Do not provision unstable annotation module for stabili…

### PR DESCRIPTION
**Pull request by yersan**
…ty levels higher than preview

Jira issue: https://issues.redhat.com/browse/WFCORE-7031

### CODE REVIEW

File: core-feature-pack/common/src/main/resources/modules/system/layers/base/org/wildfly/extension/core-management/main/module.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
         <module name="org.wildfly.extension.core-management-client"/>
         <module name="org.wildfly.security.elytron-private"/>
         <module name="org.wildfly.service"/>
-        <module name="org.wildfly.unstable.annotation.api.indexer"/>
-        <module name="org.wildfly._internal.unstable-api-annotation-index"/>
+        <module name="org.wildfly.unstable.annotation.api.indexer" optional="true"/>
```
Comment (kabir) [inline core-feature-pack/common/src/main/resources/modules/system/layers/base/org/wildfly/extension/core-management/main/module.xml:33]: I'm not totally familiar with how this works.
Does being optional AND the dependency having `jboss.stability=preview` mean it gets provisioned?

Comment (jfdenise) [reply core-feature-pack/common/src/main/resources/modules/system/layers/base/org/wildfly/extension/core-management/main/module.xml:33]: When core-management is transformed onto a Galleon package, dependencies on optional modules are ignored. An optional module is expected to be provisioned from another root. For example, a layer, a subsystem, ...

Comment (kabir) [reply core-feature-pack/common/src/main/resources/modules/system/layers/base/org/wildfly/extension/core-management/main/module.xml:33]: Looking for 'org.wildfly._internal.unstable-api-annotation-index' and 'org.wildfly.unstable.annotation.api.indexer' in the wildfly-core sources, I don't see those being referenced anywhere that could make a difference apart from  via this module dependency

Comment (yersan) [reply core-feature-pack/common/src/main/resources/modules/system/layers/base/org/wildfly/extension/core-management/main/module.xml:33]: I've added the packages via model.xml, let's see how it goes

**Comment by kabir** [comment]
@yersan Failures seem related

**Comment by jfdenise** [comment]
@yersan I think that the inclusion thanks to model.xml of these 2 modules need to be replicated in WildFly Full repo.
https://github.com/wildfly/wildfly/blob/main/ee-feature-pack/galleon-local/src/main/resources/configs/standalone/model.xml
https://github.com/wildfly/wildfly/blob/main/ee-feature-pack/galleon-local/src/main/resources/configs/domain/model.xml

**Comment by yersan** [comment]
@jfdenise @kabir I am missing something here. 

If we add packages via model.xml, are those packages provisioned when we are trimming the server using layers?

I mean, the error:

```
java.lang.AssertionError: Some expected modules have not been provisioned in test-all-layers: org.wildfly._internal.unstable-api-annotation-index,org.wildfly.unstable.annotation.api.indexer
  at org.junit.Assert.fail(Assert.java:89)
  at org.jboss.as.test.layers.LayersTest.testLayersModuleUse(LayersTest.java:172)
  at org.jboss.as.test.shared.LayersTestBase.testLayersModuleUse(LayersTestBase.java:413)
  at org.jboss.as.test.layers.base.LayersTestCase.testLayersModuleUse(LayersTestCase.java:37)
```

Looks legit to me, but I guess adding them to model.xml won't fix it in WildFly. Also, I don't get why the Galleon Jobs passed under the current situation for WildFly Core, I would expect them to fail due the above assumption where packages are not added to layers if they are declared only in model.xml

What I am missing?

**Comment by jfdenise** [comment]
@yersan , when included in standalone/model.xml, layers will benefit from them. Layers ased provisioning inherit from model.xml content.

**Comment by yersan** [comment]
Integration Jobs (until the WildFly Counterpart gets merged)

https://ci.wildfly.org/viewLog.html?buildId=462644 (Linux integration)
https://ci.wildfly.org/viewLog.html?buildId=462646 (Bootable JAR)
https://ci.wildfly.org/viewLog.html?buildId=462642 (Galleon)
https://ci.wildfly.org/viewLog.html?buildId=462648 (Microprofile)
https://ci.wildfly.org/viewLog.html?buildId=462659 (WildFly Preview)
https://ci.wildfly.org/viewLog.html?buildId=462632 (Windows)

**Comment by yersan** [comment]
This one needs https://github.com/wildfly/wildfly/pull/18287 merged first

**Comment by yersan** [comment]
https://github.com/wildfly/wildfly/pull/18287 was merged, so kicking off the integration Jobs again

**Comment by kabir** [comment]
Hi,

According to my understanding it now looks correct.

From @jfdenise  previous comment:

* The original dependencies is now optional
* Then the model.xml parts should count as being referenced from a root (and only kick in when preview or below)

**Comment by yersan** [comment]
> From @jfdenise previous comment:
> 
> The original dependencies is now optional
> Then the model.xml parts should count as being referenced from a root (and only kick in when preview or below)

Yes, this is what we did here and in WildFly with the `"optional="true" valid-for-stability="preview"` configuration on the model.xml.

@kabir based on your feedback, I assume this is approved from your side (I don't see the green tick under your review though)

**Review by kabir — APPROVED** [review]

**Comment by yersan** [comment]
Thanks @kabir @jfdenise

**Pull request merged**